cmd/prometheus/main.go

```diff
@@ -338,7 +338,7 @@ func main() {
 	a.Flag("config.auto-reload-interval", "Specifies the interval for checking and automatically reloading the Prometheus configuration file upon detecting changes.").
 		Default("30s").SetValue(&cfg.autoReloadInterval)
 
-	a.Flag("web.listen-address", "Address to listen on for UI, API, and telemetry. Can be repeated.").
+	a.Flag("web.listen-address", "Address or UNIX domain socket path to listen on for UI, API, and telemetry. Can be repeated. If set to a UNIX domain socket, --web.external-url must be set as well.").
```

Comment:
if we really want to require `--web.external-url` in this case, we'll need to do that in the code.
I don;t see why we should require it though.

Reply:
It is currently implicitly enforced in the code (the program will abort with not being able to parse the Unix socket as an external address), but I can add a more friendly error message.
We could also not have the restriction, but I figured that the external address is primarily used in the web interface and notifications, which a user generally accesses from a web browser or email client which would not have access to a local UNIX domain socket (if even supporting them in the first place) - in my case I would set the external address to the address the reverse proxy is handling Prometheus under.

Reply:
With the new changes, it is now no longer required to set `--web.external-url`. I still _recommend_ it if the user intends to access the service using a web browser. Should I document that somewhere?